apps-rendering/src/client/article.ts

```diff
@@ -87,31 +105,107 @@ function followToggle(topic: Topic): void {
 	});
 }
 
-function topicClick(e: Event): void {
-	const follow = document.querySelector('.js-follow');
+function notificationsFollowClick(e: Event): void {
+	const follow = document.querySelector('.js-follow-notifications');
 	const topic = getTopic(follow);
-
 	if (topic) {
-		followToggle(topic);
+		followToggle(
+			topic,
+			'.js-follow-notifications-status',
+			FollowNotificationStatus,
+			notificationsClient,
+		);
 	}
 }
 
-function topics(): void {
-	const follow = document.querySelector('.js-follow');
+function tagFollowClick(e: Event): void {
+	const follow = document.querySelector('.js-follow-tag');
 	const topic = getTopic(follow);
+	if (topic) {
+		followToggle(
+			topic,
+			'.js-follow-tag-status',
+			FollowTagStatus,
+			tagClient,
+		);
+	}
+}
+
+function conditionallyRenderFollowTagComponent(
+	topic: Topic,
+	followTagStatus: Element | null,
+): void {
+	const checkBridgetCompatibilty = (
+		bridgetVersion: Optional<string>,
+	): boolean =>
+		bridgetVersion
+			.map((versionString) => compare(versionString, '2.5.0', '>='))
+			.withDefault(false);
 
-	const followStatus = document.querySelector('.js-follow-status');
+	const isBridgetCompatible = getBridgetVersion().then((version) => {
+		return checkBridgetCompatibilty(version);
+	});
 
+	const isMyGuardianEnabled = environmentClient.isMyGuardianEnabled();
+
+	void Promise.all([isBridgetCompatible, isMyGuardianEnabled])
+		.then(([isBridgetCompatible, isMyGuardianEnabled]) => {
+			isBridgetCompatible &&
+				isMyGuardianEnabled &&
+				ReactDOM.render(
+					h(FollowTagStatus, {
+						isFollowing: false,
+						contributorName: topic.displayName,
+					}),
+					followTagStatus,
+				);
+		})
+		.catch((error) => {
+			logger.error(error);
+		});
+}
+
+function topics(): void {
+	const followNotifications = document.querySelector(
+		'.js-follow-notifications',
+	);
+	const followNotificationsStatus = document.querySelector(
+		'.js-follow-notifications-status',
+	);
+	const followTag = document.querySelector('.js-follow-tag');
+	const followTagStatus = document.querySelector('.js-follow-tag-status');
+
+	const topic = getTopic(followNotifications) ?? getTopic(followTag);
 	if (topic) {
-		follow?.addEventListener('click', topicClick);
+		followNotifications?.addEventListener(
+			'click',
+			notificationsFollowClick,
+		);
 		void notificationsClient.isFollowing(topic).then((following) => {
-			if (following && followStatus) {
+			if (following && followNotificationsStatus) {
 				ReactDOM.render(
-					h(FollowStatus, {
+					h(FollowNotificationStatus, {
 						isFollowing: true,
 						contributorName: topic.displayName,
 					}),
-					followStatus,
+					followNotificationsStatus,
+				);
+			}
+		});
+	}
+
+	if (topic) {
+		conditionallyRenderFollowTagComponent(topic, followTagStatus);
+		followTag?.addEventListener('click', tagFollowClick);
```

Comment:
Small point, as I think this approach would work fine, but I'm wondering whether we could encapsulate the logic below behind a switch too, since we would only want to run this if MyGuardian is enabled

Reply:
Did this becuase it was actually not rendering correctly being outside of the promise chain once we synced up on the  simulator and tested